frontend/src/Explorer/View/Dashboard/Hero.purs

```diff
@@ -33,25 +46,71 @@ heroView state =
             , P.div
                 [ P.className $ "hero-search__container" <> focusedClazz ]
                 [ P.input
-                    [ P.className $ "hero-search__input" <> focusedClazz
+                    [ P.className $ "hero-search__input hero-search__input--address-tx"
+                                  <> addrHiddenClazz
+                                  <> focusedClazz
                     , P.type_ "text"
                     , P.placeholder $ if searchInputFocused
                                       then ""
                                       else translate (I18nL.hero <<< I18nL.hrSearch) lang'
                     , P.onFocus <<< const $ DashboardFocusSearchInput true
                     , P.onBlur <<< const $ DashboardFocusSearchInput false
-                    , P.onChange $ UpdateSearchText <<< _.value <<< _.target
+                    , P.onChange $ UpdateSearchValue <<< _.value <<< _.target
                     , P.onKey "enter" $ const DashboardSearch
                     , P.value $ state ^. searchQuery
                     ]
                     []
+                , P.div
+                    [ P.className $ "hero-search__wrapper" <> epochHiddenClazz ]
+                    [ P.label
+                        [ P.className $ "hero-search__label"
+                        , P.htmlFor inputEpochName
+                        ]
+                        [ P.text $ translate (I18nL.common <<< I18nL.cEpoch) lang' ]
+                    , P.input
+                        [ P.className $ "hero-search__input hero-search__input--epoch"
```

Comment:
@akegalj That is a kind of name convention. It based of `BEM` to structure `CSS` in a well organized matter by using `Block`, `Element` and `Modifier` (http://getbem.com/naming/). That's why is called `BEM`. We do use this convention for (almost) all our CSS code. 

Here you will find a short introduction about `BEM` https://css-tricks.com/bem-101/